src/main/java/com/flowingcode/vaadin/addons/twincolgrid/TwinColGrid.java

```diff
@@ -406,16 +422,17 @@ public void setRequiredIndicatorVisible(final boolean visible) {
 	}
 
 	private void updateSelection(final Set<T> addedItems, final Set<T> removedItems) {
-		leftGridDataProvider.getItems().addAll(removedItems);
-		leftGridDataProvider.getItems().removeAll(addedItems);
-		leftGridDataProvider.refreshAll();
+		left.getItems().addAll(removedItems);
+		left.getItems().removeAll(addedItems);
+		left.getDataProvider().refreshAll();
 
-		rightGridDataProvider.getItems().addAll(addedItems);
-		rightGridDataProvider.getItems().removeAll(removedItems);
-		rightGridDataProvider.refreshAll();
+		right.getItems().addAll(addedItems);
+		right.getItems().removeAll(removedItems);
 

```

Comment:
Aqui no hay que hacer refreshAll como con la otra grid? O viendo que se hace debajo, hace falta la linea 427?

Reply:
Good catch, it's already refreshed in 434

Reply:
Fixed